server.py

```diff
@@ -21,6 +24,18 @@ def download(number):
 @app.route('/upload', method = 'POST')
 def file_upload():
     uploaded = request.files.get('upload').file.read() #uploaded outputs by user
+    u_name = request.forms.get('username') #accepting username
+
+    # storing usernames' submission along with the timestamp at which they submit it
+    global usernames
+    time = datetime.datetime.now()
+    t = time.strftime("%x") + ' ' + time.strftime("%I") + ":" + time.strftime("%M") + ":" + \
+        time.strftime("%S") + " " + time.strftime("%p")
+    if u_name not in usernames.keys():
```

Comment:
When checking for presence of something in a dictionary simply use `if u_name in usernames`. This is a hash lookup and is done in constant time. If you use `keys()` however, the lookup takes as much time as items in the dictionary.
```suggestion
    if u_name not in usernames:
```

Reply:
It might help you to read about [defaultdicts](https://docs.python.org/3.7/library/collections.html#collections.defaultdict). This line of code can be made smaller with that, although not strictly necessary.

If you don't use a default dict it's fine. In the next commit which fixes the `keys` lookup I'll merge this PR.